### PR TITLE
Add Mixed precision/LossScaler + several fixes

### PR DESCRIPTION
**Pull request by thiagocrepaldi**
This PR introduces mixed precision tests with loss scaler support

    Add input_name and loss_scale properties on LossScaler public API
    Fix CUDA training
    Add optimization_step into TrainStepInfo class
    Refactor LRSCheduler to use optimization_step instead of step
    Updated several default values at ORTTrainerOptions
    Add initial Gradient Accumulation supported. Untested
    Fix ONNX model post processing
    Refactor unit tests

### CODE REVIEW

File: orttraining/orttraining/python/training/orttrainer.py [hunk 1]
```diff
@@ -147,18 +151,27 @@ def __init__(self, model, model_desc, optim_config, loss_fn=None, options=None):
...
+
         # Set GPU device and memory limit
-        device_id = self.options.device.id
-        if 'cuda' in device_id.lower():
-            set_cuda_mem_limit(int(self.options.device.mem_limit))
-            if ':' in device_id:
-                set_cuda_device_id(int(device_id.split(':')[1]))
+        if 'cuda' in self.options.device.id.lower():
+            mem_limit = self.options.device.mem_limit
+            if  mem_limit > 0:
```
Comment (rayankrish) [inline orttraining/orttraining/python/training/orttrainer.py:169]: Should we try and warn the user if they pick something too small? If they pick 256 bytes and it doesn't crash it may be hard to debug the issue

File: orttraining/orttraining/test/python/orttraining_test_orttrainer_frontend.py [hunk 2]
```diff
@@ -589,85 +597,139 @@ def testInstantiateORTTrainer(step_fn, lr_scheduler, expected_lr_values):
...
-    # Reset the seeds
+    # Setup for the second ORTTRainer run
     torch.manual_seed(seed)
     set_seed(seed)
-
-    # Run second model train step
-    _, _, second_trainer, _, _ = generate_pytorch_transformer_model_sample(optim_config, opts, device=device_id)
-    output = second_trainer.train_step(data, targets)
+    model, _, _, _, _, _, _ = _load_pytorch_transformer_model(device)
+    second_trainer = orttrainer.ORTTrainer(model, model_desc, optim_config, loss_fn=my_loss, options=opts)
```
Comment (rayankrish) [inline orttraining/orttraining/test/python/orttraining_test_orttrainer_frontend.py:630]: Can we use the same instance of the PyTorch model? Or do a deepcopy instead?

File: orttraining/orttraining/python/training/orttrainer.py (hunk 1 above)
Comment (thiagocrepaldi) [reply orttraining/orttraining/python/training/orttrainer.py:172]: I raised a bug to the backend reporting this behavior. Let's see what they can do in this

File: orttraining/orttraining/test/python/orttraining_test_orttrainer_frontend.py (hunk 2 above)
Comment (thiagocrepaldi) [reply orttraining/orttraining/test/python/orttraining_test_orttrainer_frontend.py:641]: this is a pytorch model, which is not changed by ORT and will be used to generate an ONNX model. They can be the same as there is no risk of one run affect the other
